src/web/client/dal/fileSystemProvider.ts

```diff
@@ -310,6 +310,99 @@ export class PortalsFS implements vscode.FileSystemProvider {
         return files;
     }
 
+    async searchTextResults(query: vscode.TextSearchQuery, options: vscode.TextSearchOptions, progress: vscode.Progress<vscode.TextSearchResult>) {
+        const results = await this.searchText(
+            query,
+            { maxResults: options.maxResults }
+        );
+
+        if (results === undefined) {
+            return { limitHit: true };
+        }
+
+        for (const match of results.matches) {
+            progress.report({
+                uri: match.uri,
+                ranges: match.ranges,
+                preview: {
+                    text: match.preview,
+                    matches: match.matches,
+                },
+            });
+        }
+
+        return { limitHit: false };
+    }
+
+    async searchText(query: vscode.TextSearchQuery, options: { maxResults?: number }): Promise<ISearchQueryResults> {
+        const matches: ISearchQueryMatch[] = [];
+        const files = await this.iterateDirectory(WebExtensionContext.rootDirectory);
+
+        // counter to keep track of the number of matches
+        let counter = 0;
+        let match: ISearchQueryMatch;
+        for (const file of files) {
+            const content = await this.readFile(file);
```

Comment:
I wonder if this loop can run async such that all file reads are in parallel. This might be more performant, especially if one iteration doesn't affect another.